src/autoscaler/metricscollector/collector/collector.go

```diff
@@ -110,3 +122,23 @@ func (c *Collector) GetCollectorAppIds() []string {
 	c.lock.Unlock()
 	return appIds
 }
+
+func (c *Collector) SaveMetricsInDB() {
+	ticker := c.cclock.NewTicker(c.collectInterval)
+	metrics := make([]*models.AppInstanceMetric, 0)
+	for {
+		select {
+		case metric := <-c.dataChan:
+			metrics = append(metrics, metric)
+		case <-ticker.C():
```

Comment:
I guess we can do it more faster than collection time,  to make data available to event generator as soon as possible...

Reply:
@boyang9527 I introduced a new parameter saveInterval which is by default set to 5 secs and is also configurable.